library/src/main/java/org/apache/apex/malhar/lib/window/impl/KeyedWindowedOperatorImpl.java

```diff
@@ -145,21 +168,43 @@ public void accumulateTuple(Tuple.WindowedTuple<KeyValPair<KeyT, InputValT>> tup
       if (accum == null) {
         accum = accumulation.defaultAccumulatedValue();
       }
-      dataStorage.put(window, key, accumulation.accumulate(accum, kvData.getValue()));
+
+      InputValT inputValue = kvData.getValue();
+      AccumT newValue = accumulation.accumulate(accum, inputValue);
+      if (updatedKeyStorage != null && (earlyTriggerMillis > 0 || lateTriggerMillis > 0) && isFiringOnlyUpdatedPanes()) {
```

Comment:
What happens if both earlyTriggerMillis and lateTriggerMillis are <= 0 but earlyTriggerCount or lateTriggerCount are > 0?